src/Illuminate/View/Mix/Mix.php

```diff
@@ -0,0 +1,271 @@
+            return $this->disabledPath();
+        }
+
+        return $this->getRealPath($path, $manifestDirectory);
+    }
+
+    /**
+     * Get the path to a versioned Mix file.
+     *
+     * @param  string  $path
+     * @param  string  $manifestDirectory
+     * @return \Illuminate\Support\HtmlString
+     */
+    protected function getRealPath($path, $manifestDirectory)
+    {
+        $this->init($path, $manifestDirectory);
+
+        if ($this->hmrModeEnabled()) {
+            return $this->hmrPath();
+        }
+
+        return $this->compiledPath();
+    }
+
+    /**
+     * Set a sanitized version of assets.
+     *
+     * @param  string  $path
+     * @param  string  $manifestDirectory
+     * @return void
+     */
+    protected function init($path, $manifestDirectory)
+    {
+        $this->path = $this->sanitize($path);
+        $this->manifestDirectory = $this->sanitize($manifestDirectory);
+    }
+
+    /**
+     * Get a sanitized version of a path.
+     *
+     * @param  string  $path
+     * @return string
+     */
+    protected function sanitize($path)
+    {
+        if ($path && ! Str::startsWith($path, '/')) {
+            $path = "/{$path}";
+        }
+
+        return $path;
+    }
+
+    /**
+     * Check if the HRM mode of Mix is enabled.
+     *
+     * @return bool
+     */
+    protected function hmrModeEnabled()
+    {
+        return file_exists(public_path($this->manifestDirectory.$this->hmrFilename));
+    }
+
+    /**
+     * Get the full path to the file through the HMR server.
+     *
+     * @return \Illuminate\Support\HtmlString
+     */
+    protected function hmrPath()
+    {
+        return new HtmlString($this->hmrURI.$this->path);
+    }
+
+    /**
+     * Get the full path to the compiled file.
+     *
+     * @return \Illuminate\Support\HtmlString
+     */
+    protected function compiledPath()
+    {
+        return new HtmlString($this->manifestDirectory.$this->getPathFromManifest());
+    }
+
+    /**
+     * Get a message instead of the path when mix is disabled.
+     *
+     * @return \Illuminate\Support\HtmlString
+     */
+    protected function disabledPath()
+    {
+        return new HtmlString('Mix is disabled!');
+    }
+
+    /**
+     * Get the path from the manifest file.
+     *
+     * @return string
+     *
+     * @throws \Illuminate\View\Mix\MixException
+     */
+    protected function getPathFromManifest()
+    {
+        if (! array_key_exists($this->path, $manifest = $this->getManifest())) {
+            throw new MixException(
+                "Unable to locate Mix file: {$this->path}. Please check your ".
+                'webpack.mix.js output paths and try again.'
+            );
+        }
+
+        return $manifest[$this->path];
+    }
+
+    /**
+     * Load the manifest file.
+     *
+     * @return array
+     *
+     * @throws \Illuminate\View\Mix\MixException
+     */
+    protected function getManifest()
+    {
+        if (! $this->manifest) {
```

Comment:
No need to go further if `$this->manifest` is not empty. You can favor an early return:
```php
if ($this->manifest) {
    return $this->manifest;
}

if (! file_exists($manifestPath = public_path($this->manifestDirectory.$this->manifestFilename))) {
    throw new MixException('The Mix manifest does not exist.');
}

$this->manifest = json_decode(file_get_contents($manifestPath), true);

if (json_last_error() === JSON_ERROR_NONE) {
    return $this->manifest;
}

throw new MixException("The Mix manifest isn't a proper json file.");
```